x-pack/legacy/plugins/ingest/server/libs/adapters/policy/default.ts

```diff
@@ -39,192 +47,133 @@ export class PolicyAdapter {
-    const inputs = await this.so.bulkGet(
-      ids.map(
-        (id): SavedObjectsBulkGetObject => ({
-          id,
-          type: 'inputs',
-        })
-      )
-    );
-
-    return inputs.saved_objects.map(input => input.attributes);
+  // public async listVersions(
+  //   sharedID: string,
+  //   activeOnly = true,
+  //   page: number = 1,
+  //   perPage: number = 25
+  // ): Promise<PolicyFile[]> {
+  //   const policies = (await this.so.find<any>({
+  //     type: 'policies',
+  //     search: sharedID,
+  //     searchFields: ['shared_id'],
+  //     page,
+  //     perPage,
+  //   })).saved_objects;
+
+  //   if (!activeOnly) {
+  //     const backupPolicies = await this.so.find<BackupPolicyFile>({
+  //       type: 'backup_policies',
+  //       search: sharedID,
+  //       searchFields: ['shared_id'],
+  //     });
+  //     policies.concat(backupPolicies.saved_objects);
+  //   }
+
+  //   return policies.map<PolicyFile>(policySO => {
+  //     const policy = {
+  //       id: policySO.id,
+  //       ...policySO.attributes,
+  //     };
+  //     const decoded = RuntimePolicyFile.decode(policy);
+  //     if (isRight(decoded)) {
+  //       return decoded.right;
+  //     } else {
+  //       throw new Error(`Invalid PolicyFile data. == ${policy}`);
+  //     }
+  //   });
+  // }
+
+  public async update(
+    user: FrameworkUser,
+    id: string,
+    policy: StoredPolicy
+  ): Promise<StoredPolicy> {
+    await this.so.update<StoredPolicy>(user, 'policies', id, policy);
+
+    return policy;
   }
 
-  public async listInputsforPolicy(
-    policyId: string,
-    page: number = 1,
-    perPage: number = 25
-  ): Promise<DatasourceInput[]> {
-    const inputs = await this.so.find({
-      type: 'inputs',
-      search: policyId,
-      searchFields: ['policy_id'],
-      perPage,
-      page,
-    });
-
-    return inputs.saved_objects.map(input => input.attributes);
+  public async delete(user: FrameworkUser, id: string): Promise<void> {
+    await this.so.delete(user, 'policies', id);
   }
-
-  public async addInputs(inputs: DatasourceInput[]): Promise<string[]> {
-    const newInputs = [];
-    for (const input of inputs) {
-      newInputs.push(await this.so.create<DatasourceInput>('inputs', input));
+  // public async getBackup(id: string): Promise<BackupPolicyFile> {
+  //   const policy = await this.so.get<BackupPolicyFile>('backup_policies', id);
+
+  //   if (policy.error) {
+  //     throw new Error(policy.error.message);
+  //   }
+
+  //   if (!policy.attributes) {
+  //     throw new Error(`No backup policy found with ID of ${id}`);
+  //   }
+  //   if (isRight(RuntimePolicyFile.decode(policy.attributes))) {
+  //     return policy.attributes as BackupPolicyFile;
+  //   } else {
+  //     throw new Error(`Invalid BackupPolicyFile data. == ${policy.attributes}`);
+  //   }
+  // }
+}
+function _joinFilters(filters: string[], operator = 'AND') {
```

Comment:
I see that this is being used in `list()`, but the logic there will only ever return one filter, so using this function there is kind of moot. what kind of other use cases do we foresee where `_joinFilters` is used for more than one filter?

Reply:
We might include some things like showing active/inactive policies (using the status key) or other such listings. That said, we can always just re-add this code at that time